test/t8_cmesh/t8_gtest_cmesh_copy.cxx

```diff
@@ -25,19 +25,19 @@
 #include "t8_cmesh/t8_cmesh_trees.h"
 #include "t8_cmesh/t8_cmesh_partition.h"
 #include <t8_eclass.h>
-#include <t8_cmesh/t8_cmesh_testcases.h>
 #include <test/t8_gtest_macros.hxx>
+#include "test/t8_cmesh_generator/t8_gtest_cmesh_generator.hxx"
 
 /* Test if a cmesh is committed properly and perform the face consistency check. */
 
-class cmesh_copy_equality: public testing::TestWithParam<int> {
+class cmesh_copy_equality: public testing::TestWithParam<cmesh_generator> {
  protected:
   void
   SetUp () override
   {
-    cmesh_id = GetParam ();
-
-    cmesh_original = t8_test_create_cmesh (cmesh_id);
+    cmesh_gen = GetParam ();
+    cmesh_gen.create_cmesh ();
```

Comment:
Why not just also return the cmesh object here and store it in `cmesh_original`? Then you could save the line after it.

Reply:
Because the Range would create all cmeshes at once, leading to a huge memory consumption. That way we only create the cmesh when we need it.